setup.py

```diff
@@ -53,6 +53,8 @@
 CRASHTRACKER_DIR = HERE / "ddtrace" / "internal" / "datadog" / "profiling" / "crashtracker"
 STACK_V2_DIR = HERE / "ddtrace" / "internal" / "datadog" / "profiling" / "stack_v2"
 
+BUILD_PROFILING_NATIVE_TESTS = os.getenv("DD_PROFILING_NATIVE_TESTS", "0").lower() in ("1", "yes", "on", "true")
```

Comment:
I think this is fine for now, but we are in a weird situation where we want to make sure the same assets that are tested are the ones we distribute.  That would mean optional flags like this become necessary, and we need to decide on how we structure things.

That said, apart from running tests in a new job, with a new runner, and new configuration, I think your solution here is the only other option.

Just flagging this for awareness, I don't think there's anything to be done in this PR.